### ISSUE TITLE
Data deleted

### ISSUE THREAD

**Issue by kevindmoore** [tekartik/sqflite#61]
I've been using sqflite for a bit and have noticed that the data can disappear when I restart the app or the app crashes.

**Comment by alextekartik** [comment]
Thanks for the report, I would be interested in being able to reproduce this as I never has such issue (SQLite is know to be robust on this). The only think I can think of would be that you open the database multiple times. If you have sample code to reproduce I would be interested. I'm not sure what you mean by "the data can disappear". Do you loose your entire database or only the most recent added data?

Personally I have one global reference Database in my flutter application to avoid lock issues. Opening the database should be safe if called multiple time.

Keeping a reference only in a widget can cause issues with hot reload if the reference is lost (and the database not closed yet).

**Comment by kevindmoore** [comment]
Sorry, it was my fault. Still learning Flutter and Dart. I did learn that I can close the database in the dispose function. I also learned that I need to create an auto incrementing id, retrieve that id from the insert call and set the id in my local class. Otherwise, I can't delete or update it later.
I have written some classes that help me write out the SQL statements.
I have created an Android SQLite library that uses reflection to build the database and would like something like that (ORM type) but it looks like Flutter doesn't have the equivalent (mirrors).
